### PR TITLE
implement Add/Edit/Save/Delete in key/value pair editor

### PR DESCRIPTION
**Pull request by davidsbailey**
datamode-get-set

### CODE REVIEW

File: apps/src/applab/DataProperties.jsx [hunk 1]
```diff
@@ -72,20 +68,142 @@ const DataProperties = React.createClass({
...
+              style={dataStyles.button}
+              onClick={this.handleAdd}
+          >
+            Add pair
+          </button>
+        </td>
+      </tr>
+    );
+  }
+}));
```
Comment (Bjvanminnen) [inline apps/src/applab/DataProperties.jsx]: As components get more complicated that a simple render function, I would encourage moving them into their own file.

File: apps/src/applab/DataProperties.jsx [hunk 2]
```diff
@@ -72,20 +68,142 @@ const DataProperties = React.createClass({
...
-        <td style={dataStyles.cell}>{this.props.value}</td>
         <td style={dataStyles.cell}>
-          <button className="btn" style={dataStyles.editButton}>Edit</button>
-          <button className="btn btn-danger" style={dataStyles.button}>Delete</button>
+          {this.state.isEditing ?
+            <input
+                style={dataStyles.input}
+                value={this.state.newValue}
+                onChange={this.handleChange}
+            ></input> :
```
Comment (Bjvanminnen) [inline apps/src/applab/DataProperties.jsx]: Could just be `/>` instead of `></input>`

**Comment by Bjvanminnen** [comment]
Weird that when you click "Edit" the button becomes slightly larger (and moves over the delete button)?

**Comment by Bjvanminnen** [comment]
Generally lgtm. I'd probably start moving some of these components into their own files at this point.

**Comment by davidsbailey** [comment]
Good call. That's starting to be a lot of files, so I've gone one step further and moved all the data browser files into `apps/src/applab/dataBrowser`.

**Pull request merged**